tests/integration/records/test_records_serializers_json.py

```diff
@@ -148,21 +148,58 @@ def test_zero_citations_in_vnd_plus_inspire_record_ui_json(isolated_api_client):
                                        headers={'Accept': 'application/vnd+inspire.record.ui+json'})
     assert response.status_code == 200
     result = json.loads(response.get_data(as_text=True))
-    assert result['metadata']['citations_count'] == 0
+    assert result['metadata']['citation_count'] == 0
 
 
 def test_non_zero_citation_count_in_vnd_plus_inspire_record_ui_json(isolated_api_client):
+    from invenio_db import db
+    with db.session.begin_nested():
+        record_json = {
+            'control_number': 123,
+        }
+        record = TestRecordMetadata.create_from_kwargs(json=record_json).inspire_record
+        url = '/literature/123'
+        # Add citation
+        ref = {'control_number': 1234, 'references': [{'record': {'$ref': record._get_ref()}}]}
+        TestRecordMetadata.create_from_kwargs(json=ref)
+
+        response = isolated_api_client.get(url,
+                                           headers={'Accept': 'application/vnd+inspire.record.ui+json'})
+        assert response.status_code == 200
+        result = json.loads(response.get_data(as_text=True))
+        assert result['metadata']['citation_count'] == 1
+
+        # # Add second citation
+        # ref = {'control_number': 12341, 'references': [{'record': {'$ref': record._get_ref()}}]}
+        # TestRecordMetadata.create_from_kwargs(json=ref)
+        #
+        # response = isolated_api_client.get(url,
+        #                                    headers={'Accept': 'application/vnd+inspire.record.ui+json'})
+        # assert response.status_code == 200
+        # result = json.loads(response.get_data(as_text=True))
+        # assert result['metadata']['citation_count'] == 2
+
+
+def test_zero_citation_count_in_es(isolated_api_client):
+    cn_map = {
+        1234: 1,
+        12345: 0
+    }
     record_json = {
-        'control_number': 123,
+        'control_number': 1234,
     }
     record = TestRecordMetadata.create_from_kwargs(json=record_json).inspire_record
-    url = '/literature/123'
+    url = '/literature/?page=1&size=25&q=control_number%3A1234'
```

Comment:
you can use `/record/1234`. There is a proxy which gets by itself the collection the record belongs to and redirect to the correct page ;)
Try for example https://labs.inspirehep.net/record/1689220.

Reply:
But here we want the results from ES to check if it has citation_count inside. If we will use record/1234 the data would be pulled from DB not ES

Reply:
Oh, I see your point.
Just for you to know, there also `get_es_record` :point_right: https://github.com/inspirehep/inspire-next/blob/master/inspirehep/utils/record_getter.py#L68
Choose the one you prefer :+1: